### PR TITLE
PRO-3045: the alpha-5 changelog is a changelog, and the release gate runs

### PR DESCRIPTION
**Pull request by kdonovan**
Prep work for actually cutting alpha-5. Four things stood between us and `rake release`.

## 1. The changelog read as a dev diary

The alpha.5 section had grown to **144 bullets / 19,520 words** — 12x the total prose of the previous *five* releases combined (66 bullets / 1,570 words), median bullet 77 words against a previous max of 128, and one bullet at 2,203 words. Now **127 bullets / 8,620 words**, median 68, nothing over 175.

The structural fix matters more than the word count: labels were calibrated to intra-cycle commits rather than to alpha.4.3. `shape:` does not appear in `lib/` at tag `v0.1.0.pre.alpha.4.3` — it has never shipped — yet its section was 23 bullets / 6,111 words (31% of the release), 19 of them `[BREAKING]`/`[BUGFIX]` against states no user ever had, and **none of them introduced the feature**. That section is now one `[FEAT]` that says what `shape:` is, plus the rules an author actually hits.

Also gone: bug archaeology inside entries, scope rebuttals aimed at reviewers, spec-suite bookkeeping, a 367-word `[INTERNAL]` entry enumerating 69 privatized method names, and three CI entries written as PR descriptions (now one line).

The rationale is **deleted, not relocated**. It is already durable in the PR bodies (450–1,600 words each), the Linear tickets, and the specs; and the docs already carry the "what" for every alpha.5 surface, so entries link to anchors that exist rather than growing new docs pages. The only "why" kept is the clause that changes what a reader *does* — e.g. `sensitive: 1` previously redacted nothing, so audit those declarations.

The preamble lists the changes that **don't** announce themselves, so a consumer knows what to re-test. **No alpha-5 upgrade-notes docs page yet:** of the 50 `[BREAKING]` entries, ~14 are non-events on never-shipped surfaces and ~22 are mechanical renames that raise at load — a migration doc would be most confident where it's least needed and hedged on the behavioral dozen, where the honest instruction is "re-run your suite." Its implied completeness would be worse than nothing with as few downstream users as we have. Revisit when public.

`docs/reference/class.md` gains a stable `{#shape-blocks}` anchor so the changelog has something durable to link at.

## 2. `rake benchmark:check` could not run at all

`benchmark/support/axn_scenarios.rb` had two scenario classes (`CompositionAction`, `ComplexAction`) declaring both `step`s and a `def call` — which the steps collision guard added this cycle rejects at load. The gate crashed before measuring anything, and since `release:guard_clean` is enhanced with `benchmark:check`, `rake release` could not run.

With the empty `def call` bodies deleted the gate works — and then blocks, on 14 allocation regressions against alpha.4.3. That sent me profiling (§3), and the outcome is re-baselining (§4).

Removing that `def call` also exposed a deeper problem in the same two classes, which **Codex caught on this PR** and which the audit it prompted widened. Their inline step children declared no `expects`, so `name` wasn't a reader inside the block and every call settled as `NameError` — the two scenarios were measuring a failure path, not a composition. Two further layers of the same defect: `ComplexAction#add_metadata` read `@user_id` from the parent's `before` hook (a step runs as its own action, so a parent ivar isn't visible — it exposed nil and failed the parent's *outbound* contract), and a step's `expects` has to mirror the parent's **tolerances**, not just name the fields (declaring `age`/`admin` bare made them required in the child, where `age` is optional upstream and `admin` arrives as the parent's `false` default, which a presence check rejects as blank).

All 17 scenarios now succeed, verified individually. The two corrected ones were measuring about half their real work — `composition` 79,301 → 146,000 objects (+84%), `complex` 106,602 → 214,400 (+101%) — so the alpha.4.3-vs-alpha.5 delta for those two was meaningless in **both** directions: a failure path on each side, not the "improvement" it read as. The other 15 scenarios, and every conclusion drawn from `basic` below, are unaffected.

## 3. A dotless field read now allocates nothing, and a cross-tree benchmark run can't lie

`Extract#call` ran `field.to_s.split(".")` on **every read of every field**, for a path fixed at declaration — and since PRO-2908 routed top-level reads through the resolver, that's the hottest path in the library (six resolutions per call for a two-field action). `Symbol#name` returns the interned frozen String rather than a fresh copy, and a single-segment path needs neither the segment Array nor the reduce block:

| scenario | before | after |
|---|---|---|
| `basic` | 506 objects/call (+29.6% vs alpha.4.3) | 488 (**+25.0%**) |
| `hooks` | +29.2% | +24.7% |
| `minimal` | +17.9% | +14.7% |

No cache, deliberately: memoizing the split per field name recovered the same 18 objects and bought a process-global mutable Hash, a thread question, and an invalidation surface. An empty path still reads as the source itself, kept explicit so the shortcut can't move it.

**The benchmark harness now refuses to measure the wrong axn.** Comparing this tree against a worktree at an older ref is the technique for attributing a regression, and requiring that tree's scenarios from this checkout's bundle silently measures *this* tree — axn's internal requires are non-relative, so only the outermost `require_relative` lands in the other tree. Between nearby refs that succeeds and reports a zero delta reading as "no effect". `axn_scenarios.rb` now compares `Object.const_source_location("Axn::VERSION")` against its own root and aborts with the correct invocation, so every benchmark path inherits the check rather than each script remembering. AGENTS.md records the hazard beside the A/B recipe.

Neither gets a CHANGELOG entry: `FieldResolvers` ships for the first time in alpha-5, so "allocates less than it did last week" is invisible to every consumer, and `benchmark/` isn't in the packaged gem. Same rule this PR's rewrite applies.

The remaining **~230 objects/call** — caching the per-call ActiveModel validator class, keyed on `FieldConfig` identity rather than on option-bag content — is [PRO-3050](https://linear.app/teamshares/issue/PRO-3050/axn-low-hanging-performance-improvements), with the two spec failures that rule out the naive key written up there. Prototyped, it takes `basic` to 256 objects/call: **34% below alpha.4.3**. So the regression is incidental, not the price of the features.

## 4. Re-baselining, so the gate stops blocking the release

The gate blocks `rake release` on any machine holding the alpha.4.3 baseline, and baselines live in the gitignored `tmp/benchmark_reports/` — so no committed file can unblock it. This ships the mechanism instead:

- **`rake benchmark:accept`** records the current numbers as the baseline, for a regression that's been reviewed and written down (here: PRO-3050).
- **The release tail no longer aborts** when this version's snapshot already exists. It shared `benchmark:release`'s by-hand guard, so accepting a regression *before* releasing would have failed the rake invocation with the gem already pushed and the tag already up. It skips with a notice now; the by-hand task keeps aborting, where an existing snapshot means you're about to lose a baseline by accident.
- **CONTRIBUTING.md gains a `Releasing` section** — there wasn't one anywhere. What CI doesn't cover (`verify_async` needs a local Redis), that `gem push` needs MFA, that baselines are per-machine so you must release from the machine holding the previous one, and that accepting a regression isn't a way to quiet a red gate. (In CONTRIBUTING rather than AGENTS.md: the latter is loaded into every agent context, and a manual maintainer-only flow doesn't earn that cost.)

**Release flow:** `rake benchmark:accept` → `rake release`.

## Verification

Everything below was run, not left for review:

- **Coverage of the rewrite.** All **50** of the original `[BREAKING]` entries map onto the new section (several deliberately merged — the twelve raw-`shape:`-grammar entries became one). A diff of every backticked identifier across both versions (856 → 525) surfaces **no dropped public API**: the 335 that disappeared are internal names (the `_`-prefixed roster, `Axn::Core::`/`Internal::` paths), prose fragments, and spelling variants of APIs still present.
- **The preamble's behavioral set.** Classifying every `[BREAKING]` entry as "announces itself" vs "needs a re-test" **found six missing**: the step-exception reclassification, both `model:` read changes, the empty-container failure reason, `Serialization.render`'s new rejections, and the `Axn::Configurable` Proc-`default:` flip (silent, and in opposite directions — a Proc `default:` is now invoked on every unset read while an assigned Proc is returned as-is). All now listed; `sensitive:`/`user_facing:` moved to an audit note, since those raise at load.
- **Every benchmark scenario actually succeeds.** All 17 driven individually with the runner's own sample data, asserting `ok?` — which is how the two step scenarios' failure path was confirmed and then fixed. It is now an explicit check rather than an assumption.
- **Re-baselining, all four paths.** `benchmark:accept` records the snapshot; `benchmark:check` then reports no regression across 17 scenarios and exits 0 — which also confirms the numbers are stable run-to-run, so accept-then-check is meaningful; the release tail skips instead of aborting; by-hand `benchmark:release` still refuses to clobber. Baseline was re-accepted after the scenario fix, so it encodes the corrected numbers.
- **The load-path guard fires on the case that was silent.** Profiling a `HEAD~3` worktree from this bundle previously loaded this tree's `lib/axn/mountable/…` while running the worktree's scenarios; it now aborts.
- **`bundle exec rake spec`** — 5,095 examples, 0 failures, with the `Extract` change in place.
- **`bundle exec rubocop`** — clean on all four touched Ruby files.
- **`yarn docs:check`** — fresh build, all internal links resolve across 36 files (the standalone link checker fails against a stale `dist`; it passes on a rebuild).
- **`bundle exec rake verify_async`** — 10/10 with Redis up (no CI job covers this).
- **`bundle exec rake downstream:check`** — all 7 sibling gems' constraints admit alpha.5.

Nothing here needs re-running. What's left is judgement rather than verification: whether the rewritten entries read right to you, and whether the behavioral list is the set you'd flag.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: benchmark/support/axn_scenarios.rb [hunk 1]
```diff
@@ -208,10 +236,6 @@ class CompositionAction
       step :add_timestamp do
         expose :processed_at, Time.now
       end
-
-      def call
-        # Steps handle the logic
-      end
     end
```
[the remark this answers is not in the document]
Comment (kdonovan) [reply benchmark/support/axn_scenarios.rb:242]: [Via Opus 5] Confirmed and fixed in `4300a3a0` — thank you, this was real and it mattered more than it looks.

Probed before believing it: both scenarios settled as `ok?=false, outcome=exception` with `NameError: undefined local variable or method 'name'`. Auditing all 17 scenarios rather than just the two flagged found two further layers of the same defect:

1. `ComplexAction#add_metadata` read `@user_id` from the parent's `before` hook. A step runs as its own action, so a parent ivar isn't visible — it exposed nil and failed the parent's **outbound** contract. The id is generated inside the step now.
2. A step's `expects` has to mirror the parent's **tolerances**, not just name the fields. Declaring `age`/`admin` bare made them required in the child, where `age` is optional upstream and `admin` arrives as the parent's `false` default — which a presence check rejects as blank. Now `age: { type: Integer, optional: true }` and `admin: { type: :boolean }` (no injected presence check).

All 17 scenarios verified to succeed individually. The two corrected ones were measuring about half their real work: composition 79,301 → 146,000 objects (+84%), complex 106,602 → 214,400 (+101%). So the alpha.4.3-vs-alpha.5 delta for those two was meaningless in *both* directions — a failure path on each side, not the improvement it read as. The other 15 scenarios and every conclusion drawn from `basic` are unaffected, and the baseline has been re-accepted on the corrected scenarios.
